roadmap/implementers-guide/src/node/backing/statement-distribution.md

```diff
@@ -1,107 +1,453 @@
+        fragment trees.
+-   `NetworkBridgeTxMessage::SendRequests`
+    -   Sends requests, initiating request/response protocol.
+
+## Request/Response
+
+-   Why:
+    -   Validators do not eagerly send each other heavy
+        `CommittedCandidateReceipt`, but instead request these lazily through
+        request/response protocols.
+
+### Protocol
+
+1.  Requesting Validator
+
+    -   Requests are queued up with `RequestManager::get_or_insert`.
+        -   Done as needed, when handling incoming manifests/statements.
+    -   `RequestManager::dispatch_requests` sends any queued-up requests.
+        -   Calls `RequestManager::next_request` to completion.
+            -   Creates the `OutgoingRequest`, saves the receiver in
+                `RequestManager::pending_responses`.
+        -   Does nothing if we have more responses pending than the limit of parallel
+            requests.
+
+2.  Peer
+
+    -   Requests come in on a peer on the `IncomingRequestReceiver`.
+        -   Runs in a background responder task which feeds requests to `answer_request`
+            through `MuxedMessage`.
+        -   This responder task has a limit on the number of parallel requests.
+    -   `answer_request` on the peer takes the request and sends a response.
+        -   Does this using the response sender on the request.
+
+3.  Requesting Validator
+
+    -   `receive_response` on the original validator yields a response.
+        -   Response was sent on the request's response sender.
+        -   Uses `RequestManager::await_incoming` to await on pending responses in an
+            unordered fashion.
+        -   Runs on the `MuxedMessage` receiver.
+    -   `handle_response` handles the response.
+
+### API
+
+-   `dispatch_requests`
+    -   Dispatches pending requests for candidate data & statements.
+-   `answer_request`
+    -   Answers an incoming request for a candidate.
+    -   Takes an incoming `AttestedCandidateRequest`.
+-   `receive_response`
+    -   Wait on the next incoming response.
+    -   If there are no requests pending, this future never resolves.
+    -   Returns `UnhandledResponse`
+-   `handle_response`
+    -   Handles an incoming response.
+    -   Takes `UnhandledResponse`
+
+## Manifest
+
+-   What it is: A message about a known backed candidate, along with a description
+    of the statements backing it.
+-   Can be one of two kinds:
+    -   `Full`
+        -   Contains information about the candidate and should be sent to peers who
+            may not have the candidate yet.
+    -   `Acknowledgement`
+        -   Omit information implicit in the candidate, and should be sent to peers
+            which are guaranteed to have the candidate already.
+
+### Manifest exchange
+
+-   Occurs between us and a peer in the grid.
+-   Why:
+    -   Indicates that both nodes know the candidate as valid and backed.
+    -   Allows the nodes to send `Statement` messages directly to each other for any
```

Comment:
'need' is a bit too strong, w.r.t. previous comment about what late statements are useful for